types/cheerio/index.d.ts

```diff
@@ -13,292 +13,295 @@
-    prevUntil(element: CheerioElement, filter?: string): Cheerio;
-    prevUntil(element: Cheerio, filter?: string): Cheerio;
-
-    slice(start: number, end?: number): Cheerio;
-
-    siblings(selector?: string): Cheerio;
-
-    children(selector?: string): Cheerio;
-
-    contents(): Cheerio;
-
-    each(func: (index: number, element: CheerioElement) => any): Cheerio;
-    map(func: (index: number, element: CheerioElement) => any): Cheerio;
-
-    filter(selector: string): Cheerio;
-    filter(selection: Cheerio): Cheerio;
-    filter(element: CheerioElement): Cheerio;
-    filter(elements: CheerioElement[]): Cheerio;
-    filter(func: (index: number, element: CheerioElement) => boolean): Cheerio;
-
-    not(selector: string): Cheerio;
-    not(selection: Cheerio): Cheerio;
-    not(element: CheerioElement): Cheerio;
-    not(func: (index: number, element: CheerioElement) => boolean): Cheerio;
-
-    first(): Cheerio;
-    last(): Cheerio;
-
-    eq(index: number): Cheerio;
-
-    get(): any[];
-    get(index: number): any;
-
-    index(): number;
-    index(selector: string): number;
-    index(selection: Cheerio): number;
-
-    end(): Cheerio;
-
-    add(selectorOrHtml: string): Cheerio;
-    add(selector: string, context: Document): Cheerio;
-    add(element: CheerioElement): Cheerio;
-    add(elements: CheerioElement[]): Cheerio;
-    add(selection: Cheerio): Cheerio;
-
-    addBack(): Cheerio;
-    addBack(filter: string): Cheerio;
-
-    // Manipulation
-    appendTo(target: Cheerio): Cheerio;
-    prependTo(target: Cheerio): Cheerio;
-
-    append(content: string, ...contents: any[]): Cheerio;
-    append(content: Document, ...contents: any[]): Cheerio;
-    append(content: Document[], ...contents: any[]): Cheerio;
-    append(content: Cheerio, ...contents: any[]): Cheerio;
-
-    prepend(content: string, ...contents: any[]): Cheerio;
-    prepend(content: Document, ...contents: any[]): Cheerio;
-    prepend(content: Document[], ...contents: any[]): Cheerio;
-    prepend(content: Cheerio, ...contents: any[]): Cheerio;
-
-    after(content: string, ...contents: any[]): Cheerio;
-    after(content: Document, ...contents: any[]): Cheerio;
-    after(content: Document[], ...contents: any[]): Cheerio;
-    after(content: Cheerio, ...contents: any[]): Cheerio;
-
-    insertAfter(content: string): Cheerio;
-    insertAfter(content: Document): Cheerio;
-    insertAfter(content: Cheerio): Cheerio;
-
-    before(content: string, ...contents: any[]): Cheerio;
-    before(content: Document, ...contents: any[]): Cheerio;
-    before(content: Document[], ...contents: any[]): Cheerio;
-    before(content: Cheerio, ...contents: any[]): Cheerio;
-
-    insertBefore(content: string): Cheerio;
-    insertBefore(content: Document): Cheerio;
-    insertBefore(content: Cheerio): Cheerio;
+interface Document {}
 
-    remove(selector?: string): Cheerio;
+declare namespace cheerio {
+    interface CheerioElement {
```

Comment:
Could we just call this `Element`? `cheerio.Element` looks a bit nicer than `cheerio.CheerioElement` :-)

Reply:
Updated!

Reply:
Users tho won't be seeing the `cheerio.`, they'll be importing `Element`, which would likely be confusing.

Reply:
Personally I agree that CheerioElement is better, because there are a lot of kinds of Elements. Although nothing stops developers from importing the namespace just as before.

Reply:
Good points! My usage pattern seems different than yours and it might not be the recommended approach.

I expected usage to look like this:

```ts
const cheerio = require('cheerio')
// or 
import cheerio from 'cheerio'

function toHTML(element: cheerio.Element): string { ... }
```

Would this work?

Reply:
I think that what's _far_ more common, and what vscode's automatic importing would do for the type, is:
```jsx
import cheerio, { Element } from 'cheerio';

const $ = cheerio.load("...")

function toHTML(element: Element): string { ... }
```

Reply:
This got merged, though I'm happy to revisit this in a follow-on. The plan is to move the types upstream (cheeriojs/cheerio#1471) so perhaps the follow-on changes can happen there.